src/main/scala/com.snowplowanalytics.maxmind/iplookups/SpecializedReader.scala

```diff
@@ -0,0 +1,30 @@
+package com.snowplowanalytics.maxmind.iplookups
+
+import java.net.InetAddress
+
+import com.maxmind.geoip2.DatabaseReader
+import com.maxmind.geoip2.exception.AddressNotFoundException
+import com.snowplowanalytics.maxmind.iplookups.ReaderFunctions.ReaderFunction
+
+trait GeoIp2Db
+object Isp extends GeoIp2Db
+object Org extends GeoIp2Db
+object Domain extends GeoIp2Db
+object NetSpeed extends GeoIp2Db
```

Comment:
could we have a sealed trait and case objects?

Reply:
Ups, they're not even in use (they were in the first version of this code), I will remove these!